ecl_threads/include/ecl/threads/thread_win.hpp

```diff
@@ -451,56 +442,12 @@ Error Thread::start(void (C::*function)(), C &c, const Priority &priority, const
 	}
 
 	thread_task = new threads::ThreadTask< BoundNullaryMemberFunction<C,void> >(generateFunctionObject( function, c ), priority);
-
-    DWORD threadid;
-
-    thread_handle = CreateThread(NULL,
-    	0,
-    	(LPTHREAD_START_ROUTINE)threads::ThreadTask< BoundNullaryMemberFunction<C,void> >::EntryPoint,
-    	thread_task,
-    	0,
-    	&threadid);
-
-    if (!thread_handle) {
-    	ecl_debug_throw(StandardException(LOC, UnknownError, "Failed to create thread."));
-    	return Error(UnknownError);
-    }
-
-    BOOL bResult = FALSE;
-
-    if (priority >= RealTimePriority1) {
-    	bResult = SetThreadPriority(thread_handle, THREAD_PRIORITY_TIME_CRITICAL);
-    }
-
-    switch (priority) {
-        case CriticalPriority:
-        	bResult = SetThreadPriority(thread_handle, HIGH_PRIORITY_CLASS);
-            break;
-        case HighPriority:
-        	bResult = SetThreadPriority(thread_handle, THREAD_PRIORITY_ABOVE_NORMAL);
-            break;
-        case LowPriority:
-        	bResult = SetThreadPriority(thread_handle, THREAD_PRIORITY_BELOW_NORMAL);
-            break;
-        case BackgroundPriority:
-        	bResult = SetThreadPriority(thread_handle, THREAD_PRIORITY_IDLE);
-            break;
-        default:
-            break;
-    }
-
-    if (!bResult) {
-        ecl_debug_throw(threads::throwPriorityException(LOC));
-    }
-
-    return Error(NoError);
+    return initialise(&threads::ThreadTask< BoundNullaryMemberFunction<C,void> >::EntryPoint, priority, stack_size);
```

Comment:
Check your pointer conversions.

Reply:
looked into the use of function pointers, here this `&` before the function name does not perform address-of action; instead, it only creates reference to the function. so, the following 2 are both ok:
```c++
void f() {}

typedef void (*voidFunc) (void);

voidFunc fp1 = f; // ok
voidFunc fp2 = &f; // also ok, points to the same function as fp1
```